src/foundry/client-esm/canvas/tokens/ring.d.mts

```diff
@@ -1,5 +1,180 @@
-declare class TokenRing {}
+/**
+ * Dynamic Token Ring Manager.
+ */
+declare class TokenRing {
+  #TokenRing: true;
 
-declare namespace TokenRing {}
+  /** A `TokenRing` is constructed by providing a reference to a `Token` object. */
+  constructor(token: Token.ConfiguredInstance);
+
+  /** The effects which could be applied to a token ring (using bitwise operations) */
+  static effects: TokenRing.Effects;
+
+  /** Is the token rings framework enabled? Will be `null` if the system hasn't initialized yet. */
+  static get initialized(): boolean | null;
```

Comment:
Is this initialized on a certain hook? If so you can use the existing `AssumeHook` helper framework. I should probably have that moved to utils.